packages/sd-jwt/src/SdJwtService.ts

```diff
@@ -0,0 +1,340 @@
+
+  private async resolveDidUrl(agentContext: AgentContext, didUrl: string) {
+    const didResolver = agentContext.dependencyManager.resolve(DidResolverService)
+    const didDocument = await didResolver.resolveDidDocument(agentContext, didUrl)
+
+    return { verificationMethod: didDocument.dereferenceKey(didUrl), didDocument }
+  }
+
+  private get hasher(): HasherAndAlgorithm {
+    return {
+      algorithm: HasherAlgorithm.Sha256,
+      hasher: (input: string) => {
+        const serializedInput = TypedArrayEncoder.fromString(input)
+        const hash = Hasher.hash(serializedInput, 'sha2-256')
+        return TypedArrayEncoder.toBase64URL(hash)
+      },
+    }
+  }
+
+  /**
+   * @todo validate the JWT header (alg)
+   */
+  private signer<Header extends Record<string, unknown> = Record<string, unknown>>(
+    agentContext: AgentContext,
+    key: Key
+  ): Signer<Header> {
+    return async (input: string) => agentContext.wallet.sign({ key, data: TypedArrayEncoder.fromString(input) })
+  }
+
+  /**
+   * @todo validate the JWT header (alg)
+   */
+  private verifier<Header extends Record<string, unknown> = Record<string, unknown>>(
+    agentContext: AgentContext,
+    signerKey: Key
+  ): Verifier<Header> {
+    return async ({ message, signature, publicKeyJwk }) => {
+      let key = signerKey
+
+      if (publicKeyJwk) {
+        if (!('kty' in publicKeyJwk)) {
+          throw new SdJwtError(
+            'Key type (kty) claim could not be found in the JWK of the confirmation (cnf) claim. Only JWK is supported right now'
+          )
+        }
+
+        const jwk = getJwkFromJson(publicKeyJwk as JwkJson)
+        key = Key.fromPublicKey(jwk.publicKey, jwk.keyType)
+      }
+
+      return await agentContext.wallet.verify({
+        signature: Buffer.from(signature),
+        key: key,
+        data: TypedArrayEncoder.fromString(message),
+      })
+    }
+  }
+
+  public async create<Payload extends Record<string, unknown> = Record<string, unknown>>(
+    agentContext: AgentContext,
+    payload: Payload,
+    {
+      issuerDidUrl,
+      holderDidUrl,
+      disclosureFrame,
+      hashingAlgorithm = 'sha2-256',
+      issuerOverrideJsonWebAlgorithm,
+    }: SdJwtCreateOptions<Payload>
+  ): Promise<{ sdJwtRecord: SdJwtRecord; compact: string }> {
+    if (hashingAlgorithm !== 'sha2-256') {
+      throw new SdJwtError(`Unsupported hashing algorithm used: ${hashingAlgorithm}`)
+    }
+
+    const issuerKeyId = issuerDidUrl.split('#')[1]
+    if (!issuerKeyId) {
+      throw new SdJwtError(
+        `issuer did url '${issuerDidUrl}' does not contain a '#'. Unable to derive key from did document`
+      )
+    }
+
+    const { verificationMethod: issuerVerificationMethod, didDocument: issuerDidDocument } = await this.resolveDidUrl(
+      agentContext,
+      issuerDidUrl
+    )
+    const issuerKey = getKeyFromVerificationMethod(issuerVerificationMethod)
+    const alg = getJwaFromKey(issuerKey, issuerOverrideJsonWebAlgorithm)
```

Comment:
```suggestion
    const issuerKey = getKeyFromVerificationMethod(issuerVerificationMethod)
    const issuerJwa = jwa /* function input parameter */ ?? getJwkFromKey(issuerKey).supportedSigningAlgorithms[0]
    const alg = getJwaFromKey(issuerKey, issuerOverrideJsonWebAlgorithm)
```

Reply:
I really would like to avoid defining which Jwa alg is associated with which key type. I think it should be key -> jwk -> jwa's associated with jwk

Reply:
I can do it inline, but `getJwaFromKey` just does the same thing.

Reply:
Not when I checked this afternoon? It was still the harcoded if keyType === Ed25519 return 'Eddsa' kinda method